build-system/git.js

```diff
@@ -84,29 +87,17 @@ exports.gitDiffStatMaster = function() {
 
 /**
  * Returns a detailed log of commits included in a PR check, starting with (and
- * including) the branch point off of master. Limited to at most 100 commits to
- * keep the output sane.
+ * including) the branch point off of master. Limited to commits in the past
+ * 30 days to keep the output sane.
  *
  * @return {string}
  */
 exports.gitDiffCommitLog = function() {
-  const branchPoint = exports.gitMergeBaseMaster();
-  let commitLog = getStdout(`git -c color.ui=always log --graph \
+  const branchCreationPoint = exports.gitBranchCreationPoint();
+  const commitLog = getStdout(`git -c color.ui=always log --graph \
 --pretty=format:"%C(red)%h%C(reset) %C(bold cyan)%an%C(reset) \
 -%C(yellow)%d%C(reset) %C(reset)%s%C(reset) %C(green)(%cr)%C(reset)" \
---abbrev-commit ${branchPoint}^...HEAD \
---max-count=${commitLogMaxCount}`).trim();
-  if (commitLog.split('\n').length >= commitLogMaxCount) {
-    commitLog += `\n${colors.yellow('WARNING:')} Commit log is longer than \
-${colors.cyan(commitLogMaxCount)} commits. \
-Branch ${colors.cyan(exports.gitBranchName())} may not have been forked from \
-${colors.cyan('master')}.`;
-    commitLog += `\n${colors.yellow('WARNING:')} See \
-${colors.cyan(
-  'https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md'
-)} \
-for how to fix this.`;
-  }
+--abbrev-commit ${branchCreationPoint}^...HEAD --since "30 days ago"`).trim();
```

Comment:
Will this work? Travis runs `git clone … --depth=50` so it should have at most 50 commit log messages available

Reply:
Yes, this works. `git log` on Travis can return as many commits as you want, all the way to the first commit.